File: docs/breez-sdk/snippets/react-native/prepare_payment_link.ts
import {
  CrossChainRouteFilter,
  InputType_Tags,
  type BreezSdk
} from '@breeztech/breez-sdk-spark-react-native'

const preparePaymentLinkViaCashapp = async (sdk: BreezSdk) => {
  // ANCHOR: prepare-payment-link-cashapp
  // Parse the recipient's external-chain address (EVM/Solana/Tron).
  const input = '<recipient address>'
  const parsed = await sdk.parse(input)
  if (parsed.tag !== InputType_Tags.CrossChainAddress) {
    throw new Error('Not a cross-chain address')
  }
  const addressDetails = parsed.inner[0]

  // List the stablecoin destinations you can send to and pick one, e.g. USDC
  // on Base. These routes are funded by an external rail, not the Spark wallet.
  // Restrict to Lightning-fundable routes.
  const routes = await sdk.getCrossChainRoutes(
    new CrossChainRouteFilter.PaymentLink({
      addressDetails
    })
  )
  const route = routes.find((r) => r.asset === 'USDC' && r.chain === 'base')
  if (route === undefined) {
    throw new Error('No USDC route on Base')
  }

  // Send $10 of USDC, funded by Cash App over Lightning. The amount is in USD
  // base units (6-decimal), so 10_000_000 = $10.00.
  const response = await sdk.preparePaymentLink({
    address: addressDetails.address,
    route,
    amount: BigInt(10_000_000),
    feePolicy: undefined,
    maxSlippageBps: undefined
  })

  // Open this Cash App URL to pay; the recipient then receives the stablecoin.
  console.log(`Open this URL in Cash App: ${response.url}`)
  console.log(`Recipient receives ~${response.estimatedOut} ${response.asset}`)
  // ANCHOR_END: prepare-payment-link-cashapp
}

File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/PreparePaymentLink.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*
import com.ionspin.kotlin.bignum.integer.BigInteger

class PreparePaymentLink {
    suspend fun preparePaymentLinkViaCashapp(sdk: BreezSdk) {
        // ANCHOR: prepare-payment-link-cashapp
        // Parse the recipient's external-chain address (EVM/Solana/Tron).
        val input = "<recipient address>"
        val parsed = sdk.parse(input)
        if (parsed !is InputType.CrossChainAddress) {
            throw IllegalArgumentException("Not a cross-chain address")
        }
        val addressDetails = parsed.v1

        // List the stablecoin destinations you can send to and pick one, e.g. USDC
        // on Base. These routes are funded by an external rail, not the Spark wallet.
        // Restrict to Lightning-fundable routes, since Cash App pays over Lightning.
        val routes = sdk.getCrossChainRoutes(
            CrossChainRouteFilter.PaymentLink(
                addressDetails = addressDetails,
            )
        )
        val route = routes.find { it.asset == "USDC" && it.chain == "base" }
            ?: throw IllegalArgumentException("No USDC route on Base")

        // Send $10 of USDC, funded by Cash App over Lightning. The amount is in USD
        // base units (6-decimal), so 10_000_000 = $10.00.
        val request = PreparePaymentLinkRequest(
            address = addressDetails.address,
            route = route,
            amount = BigInteger.fromLong(10_000_000L),
            feePolicy = null,
            maxSlippageBps = null,
        )

        val response = sdk.preparePaymentLink(request)

        // Open this Cash App URL to pay; the recipient then receives the stablecoin.
        // Log.v("Breez", "Open this URL in Cash App: ${response.url}")
        // Log.v("Breez", "Recipient receives ~${response.estimatedOut} ${response.asset}")
        // ANCHOR_END: prepare-payment-link-cashapp
    }
}
